### ISSUE TITLE
Symbol from an object as Class method error

### ISSUE THREAD

**Issue by deepsweet** [babel/babel#970]
hi.

seems like `babel@4.6.x` -> `babel@4.7.1` has some regression bug, because that code worked fine on previous version:

```js
const obj = {
  symbol: Symbol()
};

class Foo {
    [obj.symbol]() {
        console.log(123);
    }
}

const foo = new Foo();

foo[obj.symbol]();
```

```sh
├── babel@4.7.1
├── babel-core@4.7.1
├── babel-runtime@4.7.1
```

```sh
babel test.js -o test.es5.js -i runtime
```

```sh
test/node_modules/babel/lib/babel/helpers/parse.js:54
    throw err;
          ^
RangeError: test.js: Maximum call stack size exceeded
    at new Number (native)
    at t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:473:19)
    at Array.map (native)
    at Object.t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:480:19)
    at t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:478:17)
    at Array.map (native)
    at Object.t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:480:19)
    at t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:478:17)
    at Array.map (native)
    at Object.t.cloneDeep (test/node_modules/babel/lib/babel/types/index.js:480:19)
```

but this is still working:

```js
const symbol = Symbol();

class Foo {
    [symbol]() {
        console.log(123);
    }
}

const foo = new Foo();

foo[symbol]();
```

**Comment by sebmck** [comment]
Duplicate of #962.

**Comment by sebmck** [comment]
Fixed as of 4.7.2, thanks!
